### PR TITLE
PowerManagement: Initial commit

### PR DESCRIPTION
**Pull request by thopiekar**
The provider for Linux shall work, but further tests are needed for Windows and mostly for the OSX provider.

### CODE REVIEW

File: plugins/PowerManagement/PowerManagementLinux.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+'''
+Created on 10.01.2017
+
+@author: thopiekar
+'''
+
+import dbus
```
Comment (awhiemstra) [inline plugins/PowerManagement/PowerManagementLinux.py]: I would personally prefer to use Qt's DBus module instead, so we only need one DBus implementation.

Comment (thopiekar) [inline plugins/PowerManagement/PowerManagementLinux.py]: Is that really necessary?

File: plugins/PowerManagement/PowerManagementLinux.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                return
+            except:
+                del __init_api_list__[0]
+        
+        Logger.log("e", "We were not able to provide any power management provider!")
+        return
+    
+    def __init_api_solid__(self):
+            self.solid_session = None
+            devobj = dbus.SessionBus().get_object("org.kde.kded5",
```
Comment (thopiekar) [inline plugins/PowerManagement/PowerManagementLinux.py]: But the freedesktop interface doesn't allow to pass the application icon, afaik.
I couldn't find a way to pass the icon at the moment, but I'm planning to find it out later.

**Comment by thopiekar** [comment]
Well, I might be useful when sending very fat prints to a UM3 and the user just closes the lid.

In the future I'm planning to bring up a new plugin, but I'm not allowed to share any details in public at the moment. I can come over later and explain.

File: plugins/PowerManagement/PowerManagementLinux.py (hunk 1 above)
Comment (Ghostkeeper) [inline plugins/PowerManagement/PowerManagementLinux.py]: It removes a dependency with relative ease. Would be very nice. Our build system is already quite complicated, as you know.

**Comment by thopiekar** [comment]
Created an own repo for it.

**Pull request closed without merging**